packages/validation/src/types.ts

```diff
@@ -2,35 +2,79 @@ import type {
   Ajv,
   AnySchema,
   AsyncFormatDefinition,
+  Format,
   FormatDefinition,
 } from 'ajv';
 
 type Prettify<T> = {
   [K in keyof T]: T[K];
 } & {};
 
+/**
+ * Options to customize the JSON Schema validation.
+ *
+ * @param payload - The data to validate.
+ * @param schema - The JSON schema for validation.
+ * @param envelope - Optional JMESPATH expression to use as envelope for the payload.
+ * @param formats - Optional formats for validation.
+ * @param externalRefs - Optional external references for validation.
+ * @param ajv - Optional Ajv instance to use for validation, if not provided a new instance will be created.
+ */
 type ValidateParams = {
+  /**
+   * The data to validate.
+   */
   payload: unknown;
+  /**
+   * The JSON schema for validation.
+   */
   schema: AnySchema;
+  /**
+   * Optional JMESPATH expression to use as envelope for the payload.
+   */
   envelope?: string;
+  /**
+   * Optional formats for validation.
+   */
   formats?: Record<
     string,
-    | string
+    Format
+    /* | string
```

Comment:
huh? what's that?

Reply:
removed it, it was a leftover from a previous iteration. The final type is `Format`.